src/aap_eda/services/ruleset/activation_kubernetes.py

```diff
@@ -338,32 +349,34 @@ def run_activation_job(
                     self.delete_secret(
                         secret_name=secret_name,
                         namespace=namespace,
-                        job_name=job_name,
                     )
+                    logger.info(f"Secret: {secret_name} is deleted.")
                 # remove service(s) if created
                 self.delete_services(
                     namespace=namespace,
                     job_name=job_name,
                 )
+                logger.info(f"Service: {job_name} is deleted.")
 
-    def delete_secret(self, secret_name, namespace, job_name) -> None:
-        # wait until job is done
-        while True:
-            ret = client.BatchV1Api().list_namespaced_job(
+    def delete_secret(self, secret_name, namespace) -> None:
+        secrets = self.client_api.list_namespaced_secret(
+            namespace=namespace,
+            label_selector="aap=eda",
+        )
+
+        if secrets and secrets.items:
```

Comment:
If we call this function we expect to delete the secret. So if the secret is not found I feel we should have also a log record indicating it in case we should not raise an exception.